feat: Allow cache OPTIONS to specify USERNAME

joachimBurket opened the pull request:
resolves #656

Commit:
Passing username to redis arguments

WisdomPill commented on README.rst:
```diff
@@ -106,18 +106,20 @@ There are several ways to specify a database number:
     CACHES = {
         "default": {
             "BACKEND": "django_redis.cache.RedisCache",
-            "LOCATION": "redis://django@localhost:6379/0",
```
I would keep both examples if they work

joachimBurket replied:
I merged the example in which the username is passed in the URL and the password in the OPTIONS with the example of a URL unsafe password, because I thought it was redundant. Tell me if it suits you as done in [039432c](https://github.com/jazzband/django-redis/pull/657/commits/039432cd21748aa79dcc32541b657c289f74c306)

WisdomPill reviewed (changes requested):
thanks for the contribution! it would be nice to have a test for this and a change log file

Commit:
added the changelog file

Commit:
Updated README documentation

- merged examples

Commit:
added connection options test

joachimBurket commented:
I added the changelog and a test. I wasn't sure where to test the connection params so I renamed the `test_connection_string.py` file in `test_connection_params.py` and added it there with the test of the URL.

WisdomPill commented:
I would love to have tests to connect to a server with credentials

joachimBurket committed:
Merge remote-tracking branch 'upstream/master' into feat/pass-username-to-redis

joachimBurket commented:
Yeaaaah it's been a while sorry :sweat_smile:
@WisdomPill I'm not sure how the missing tests should be implemented, could you add some precision on  this?

terencehonles committed:
Merge branch 'master' into feat/pass-username-to-redis

terencehonles committed:
update readme

terencehonles commented:
> I would love to have tests to connect to a server with credentials

This would require we have a test setup with creds, but since we're just calling `ConnectionPool.from_url` and merging it with the ``OPTIONS`` this seems like a somewhat low value test. I'm adding a test that just ensures the expected kwargs are created and this is mostly the same as the tests you added except it will include resolving the precedence between the two configuration sources.

terencehonles committed:
update tests

terencehonles reviewed (approved):
LGTM, thanks
